File: C++/Problems/Problems/problem_2515.cpp
#include "problem_2515.h"

bool problem_2515::test() {
    std::vector<std::string> words { "hello", "i", "am", "leetcode", "hello" };
    std::string target = "hello";
    int startIndex = 1;

    int expected = 1;

    auto result = closestTarget(words, target, startIndex);

    std::cout << result << std::endl;

    return result == expected;
}

int problem_2515::closestTarget(std::vector<std::string> &words, std::string target, int startIndex) {
    int n = words.size();
    int result = INT_MAX;
    for (int i = 0; i < n; ++i)
    {
        int cur = (startIndex + i) % n;
        if (words[cur] == target)
            result = std::min(result, std::min(i, n - i));
    }

    return result == INT_MAX ? -1 : result;
}

File: Python/Problems/Problems/problem2515.py
import heapq
import math
import sys

from utils import print1DArray, print2DArray
from typing import List, Set
from collections import Counter

from .problem import Problem


class Problem2515(Problem):
    def test(self) -> bool:
        words = ["hello", "i", "am", "leetcode", "hello"]
        target = "hello"
        startIndex = 1

        expected = 1

        result = self.closestTarget(words, target, startIndex)

        print(result)

        return result == expected

    def closestTarget(self, words: List[str], target: str, startIndex: int) -> int:
        n = len(words)
        result = sys.maxsize
        for i in range(n):
            cur = (startIndex + i) % n
            if words[cur] == target:
                result = min(result, min(i, n - i))

        return -1 if result == sys.maxsize else result